crates/bevy_ecs/src/schedule/mod.rs

```diff
@@ -105,7 +105,7 @@ impl Schedule {
             .stage_order
             .iter()
             .enumerate()
-            .find(|(_i, stage_label)| &***stage_label == target)
+            .find(|(_i, stage_label)| stage_label.dyn_clone() == target.dyn_clone())
```

Comment:
`DynEq` won't work when comparing `Box<dyn StageLabel>` and `&dyn StageLabel` otherwise. I wasn't able to find any better way to fix this.

Reply:
the `dyn_clone` may add some overhead, but it shouldn't be much as labels probably won't be big, and it's only during stage setup

Reply:
Could you add a test somewhere that would fail with a `Box<dyn StageLabel>` without this change? As it doesn't fail on compile, it would make sure we don't change back later on

Reply:
This could work for the comparison:
```.find(|(_i, stage_label)| stage_label.as_ref() == target)```

Reply:
```suggestion
            .find(|(_i, stage_label)| stage_label.as_ref() == target)
```

Reply:
I've added the tests. @blaind, unfortunately, your suggestion didn't work :( It looks def cleaner, but my tests fail if I try it this way.